index.js

```diff
@@ -37,19 +38,33 @@ export default async function pWaitFor(condition, options = {}) {
 		}
 	});
 
-	if (timeout !== Number.POSITIVE_INFINITY) {
-		try {
-			return await pTimeout(promise, timeout);
-		} catch (error) {
-			if (retryTimeout) {
-				clearTimeout(retryTimeout);
-			}
+	if (timeout === Number.POSITIVE_INFINITY) {
+		return promise;
+	}
 
-			throw error;
+	const milliseconds = timeout;
+
+	const {
+		message: timeoutMessage,
+		milliseconds: timeoutMs,
+		customTimers
```

Comment:
I don’t get why it’s renaming the variables here only to rename them back in the options object. 

This should be a raw: 

``` js
const timeoutOptions = timeout === number ? {timeout} : options
```
Nothing more. 

I don’t have time to review the rest but overall this looks like too much unnecessary change in every part.